crud/common/utils.lua

```diff
@@ -251,11 +254,57 @@ function utils.is_uuid(value)
     return ffi.istype(uuid_t, value)
 end
 
-function utils.format_result(rows, space)
-    return {
-        metadata = table.copy(space:format()),
-        rows = rows,
-    }
+local function get_field_metadata(full_metadata, field_name)
+    dev_checks('table', 'string')
+
+    local metadata = metadata_cache[full_metadata]
+    if metadata ~= nil then
+        return metadata[field_name]
+    end
+
+    metadata_cache[full_metadata] = {}
+    for _, field in ipairs(full_metadata) do
+        metadata_cache[full_metadata][field.name] = field
+    end
+
+    return metadata_cache[full_metadata][field_name]
+end
+
+local function filter_format_fields(full_metadata, fields)
+    dev_checks('table', 'table')
+
+    local metadata = {}
+
+    for i, field in ipairs(fields) do
+        metadata[i] = get_field_metadata(full_metadata, field)
```

Comment:
What is metadata here? 
`filter_format_fields` function accepts `(space_format, field_names)`.

`get_field_metadata` is `get_field_format` in fact. And `metadata_cache` is `space_format_cache`.

I mean that word "metadata" should exists only in `format_result` function, whet we pass *space format* as *metadata*, but all these functions know nothing about metadata, they operate with *space format*.